util/db/cluster.go

```diff
@@ -232,6 +232,34 @@ func (db *db) GetProjectClusters(ctx context.Context, project string) ([]*appv1.
 	return res, nil
 }
 
+func (db *db) GetClusterServersByName(ctx context.Context, name string) ([]string, error) {
+	informer, err := db.settingsMgr.GetSecretsInformer()
+	if err != nil {
+		return nil, err
+	}
+
+	// if local cluster name is not overridden and specified name is local cluster name, return local cluster server
+	localClusterSecrets, err := informer.GetIndexer().ByIndex(settings.ByClusterURLIndexer, appv1.KubernetesInternalAPIServerAddr)
```

Comment:
Can we check `db.getLocalCluster().Name == name` first and fetch the local cluster index later? I think it is not necessary to do this if the specified name is not local

Reply:
This check is required because the local cluster can be renamed. So we cannot assume it is named "in-cluster" and have to check cluster secrets.